lib/actions/serverSetupActions.js

```diff
@@ -402,6 +417,33 @@ export function applyServer() {
     };
 }
 
+export function applyServerAgain() {
```

Comment:
This name does not clearly describe what the function does. How about `resetAndApplyServer()`? Perhaps also add some jsdoc for this function. Future developers might have problems understanding why reopening the adapter is necessary.